src/prefect/deployments/base.py

```diff
@@ -440,7 +440,21 @@ def _save_deployment_to_prefect_file(
         if deployments is None:
             parsed_prefect_file_contents["deployments"] = [deployment]
         else:
-            deployments.append(deployment)
+            for i, existing_deployment in enumerate(deployments):
+                if existing_deployment.get("name") == deployment.get("name") and (
+                    existing_deployment.get("entrypoint")
+                    == deployment.get("entrypoint")
+                ):
+                    get_logger().info(
```

Comment:
We should use `app.console.print` to match the styling of other messages.

Reply:
In `_find_flow_functions_in_file` we also use `get_logger()` since this isn't a CLI command. Should we treat this function differently just because it's not raising?

Reply:
I'm going to move this logic to deploy.py

Reply:
I've been using the logger only when `PREFECT_DBUG_MODE` is turned on. Otherwise I think we should try to use the `console`. You could pass in a `print_func` to allow the caller to control where the message is printed.